[stable-2.16] ansible-test - Add RHEL 10.0 remote (#85189)

mattclay opened the pull request:
##### SUMMARY

Backport of #85189

(cherry picked from commit 53b0f1645bfc1d83f3042e85cf54d22f1a7d4863)

##### ISSUE TYPE

Feature Request

Akasurde committed:
Enable vendored copy of distutils in entry_point test (#82232)

* Enable use of vendored copy of distutils by removing
  the environment variable SETUPTOOLS_USE_DISTUTILS

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>
(cherry picked from commit 1a759659fee56a8b7896948b55a1049f1eebce43)

mattclay committed:
ansible-test - Add RHEL 10.0 remote (#85189)

* ansible-test - Add RHEL 10.0 remote

* Update tests for RHEL 10
* Fix iptables test
* Rework rpm_key integration test

(cherry picked from commit 53b0f1645bfc1d83f3042e85cf54d22f1a7d4863)

mattclay committed:
Remove unsupported sub_key tests

mattclay committed:
Add RHEL 10 support to dnf test

mattclay committed:
Adjust signature check in tests

mkrizek commented on test/integration/targets/copy/tasks/acls.yml:
```diff
@@ -3,6 +3,13 @@
+    register: setup_acl
+
+  - name: Install the acl package on RedHat
+    dnf:
```
```suggestion
    yum:
```

mkrizek commented on test/integration/targets/copy/tasks/acls.yml:
```diff
@@ -30,6 +37,18 @@
+      when: setup_acl is changed and ansible_distribution in ('Ubuntu')
+
+    - name: Remove the acl package on RedHat
+      dnf:
```
```suggestion
      yum:
```

mkrizek committed:
use yum

The dnf action plugin is not able to redirect to yum (the value of `pkg_mgr` in RHEL 7) in 2.16 before yum removal.

mattclay committed:
Use redhat-*-fonts for yum test

The lohit-*-fonts are not available on RHEL 10.

mkrizek committed:
yum integration tests fixes

* use fonts package available in "all" RHEL versions
* use dnf Python module to retrieve the releasever value

mkrizek committed:
fix sanity

mkrizek committed:
zlib not available on RHEL10
